### PR TITLE
Create a new capable-types crate

### PR DESCRIPTION
**Pull request by jcape**
- Create mc-sgx-capable-types
- Expose mc-sgx-core-types::{Error, Result}

### CODE REVIEW

**Comment by jcape** [comment]
Current dependencies on/for this PR:
* main
  * **PR #94** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/94" target="_blank">Graphite</a> 
    * **PR #95** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/95" target="_blank">Graphite</a> 
      * **PR #96** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/96" target="_blank">Graphite</a> 
        * **PR #97** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/97" target="_blank">Graphite</a> 
          * **PR #98** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/98" target="_blank">Graphite</a> 
            * **PR #99** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/99" target="_blank">Graphite</a> 
              * **PR #101** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/101" target="_blank">Graphite</a>  👈
              * **PR #100** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/100" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/101?utm_source=stack-comment).

**Review by nick-mobilecoin — APPROVED** [review]

File: capable/types/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2022 MobileCoin Foundation
+
+#![doc = include_str!("../README.md")]
+#![no_std]
+#![deny(missing_docs, missing_debug_implementations, unsafe_code)]
```
[the remark this answers is not in the document]
Comment (jcape) [reply capable/types/src/lib.rs:5]: No, that's all intentional :-)

File: capable/types/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    SciAvailable = sgx_device_status_t::SGX_DISABLED_SCI_AVAILABLE.0 as isize,
+    /// SGX must be enabled in BIOS settings
+    ManualEnable = sgx_device_status_t::SGX_DISABLED_MANUAL_ENABLE.0 as isize,
+    /// Hyper-V must be disabled before SGX can be enabled
+    HyperVEnabled = sgx_device_status_t::SGX_DISABLED_HYPERV_ENABLED.0 as isize,
+    /// The running OS does not support enabling SGX through UEFI
+    LegacyOs = sgx_device_status_t::SGX_DISABLED_LEGACY_OS.0 as isize,
+    /// This CPU does not support SGX
+    UnsupportedCpu = sgx_device_status_t::SGX_DISABLED_UNSUPPORTED_CPU.0 as isize,
+    /// SGX must be enabled in BIOS settings
```
[the remark this answers is not in the document]
Comment (jcape) [reply capable/types/src/lib.rs:34]: Yeah, this is based on the [Intel SGX Developer Reference](https://download.01.org/intel-sgx/sgx-linux/2.17.1/docs/Intel_SGX_Developer_Reference_Linux_2.17.1_Open_Source.pdf#%5B%7B%22num%22%3A280%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C94.5%2C543%2C0%5D). The short answer is that while it's true the code in the SDK can't read UEFI variables to see how to enable SGX, the setting to turn it on is still in BIOS...

File: capable/types/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    Disabled = sgx_device_status_t::SGX_DISABLED.0 as isize,
+    /// Administrator privileges are required to read and set EFI variables
+    NoPrivilege = -2,
+}
+
+impl From<SgxError> for Error {
+    fn from(err: SgxError) -> Self {
+        match err {
+            SgxError::NoPrivilege => Error::NoPrivilege,
```
[the remark this answers is not in the document]
Comment (jcape) [reply capable/types/src/lib.rs:43]: I'm going to update these things to be `Sgx(SgxError)`.

File: capable/types/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (nick-mobilecoin) [reply capable/types/src/lib.rs:5]: oh I read that wrong I was thinking that was allow nvm

**Review by awygle — APPROVED** [review]
LGTM

**Pull request merged**